File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Inheritance/OverridingPropertyObservers.swift
@main
struct MyMain {

    static func main() {
        let someVehicle = Vehicle()
        let automatic = AutomaticCar()
        automatic.currentSpeed = 35.0
        let des = automatic.description
    }
}
class Vehicle {
    var currentSpeed = 0.0
    var description: String {
        return "traveling at \(currentSpeed) miles per hour"
    }
    func makeNoise() {
        // do nothing - an arbitrary vehicle doesn't necessarily make a noise
    }
}

class AutomaticCar: Car {
    override var currentSpeed: Double {
        didSet {
            gear = Int(currentSpeed / 10.0) + 1
        }
    }
}

//If you provide a setter as part of a property override, you must also
//provide a getter for that override. If you don’t want to modify
//the inherited property’s value within the overriding getter, you
//can simply pass through the inherited value by returning
//super.someProperty from the getter, where someProperty is the name
//of the property you are overriding.
class Car: Vehicle {
    var gear = 1
    override var description: String {
        return super.description + " in gear \(gear)"
    }
}

File: AutoTranspiledTests/try-0/OverridingPropertyObservers.cpp

#include <iostream>
using namespace std;

class Vehicle{
    public: 
        double currentSpeed =0.0;
        string description(){
            return "traveling at " + to_string(currentSpeed) + " miles per hour";
        }
        void makeNoise(){
            // Do nothing — an arbitrary vehicle doesn't necessarily make a noise
        }
};

class Car : public Vehicle {
    public: 
        int gear = 1;
        string description(){
            return Vehicle::description() + " in gear " + to_string(gear);
        }
};

class AutomaticCar : public Car {
    public: 
        double currentSpeed;
        void setCurrentSpeed(double speed){
            currentSpeed = speed;
            gear = (int) (speed / 10.0) +1;
        }
};

int main(){
    Vehicle* someVehicle = new Vehicle();
    AutomaticCar* automatic = new AutomaticCar();
    automatic->setCurrentSpeed(35.0);
    string des = automatic->description();
    cout << des;
    return 0;
}